src/fileio.cpp

```diff
@@ -336,57 +349,35 @@ bool RcdFileReader::GetBlob(void *address, size_t length)
  * @param path Path of the directory.
  * @todo At the time of writing (2021-06-30) this is tested only on Linux. Before using it anywhere else, test this on all platforms (especially Windows).
  */
-void MakeDirectory(std::string path)
+void MakeDirectory(const std::string &path)
 {
-	if (path.empty() || PathIsDirectory(path.c_str())) return;
-
-	/* Strip trailing path separators. */
-	while (StrEndsWith(path.c_str(), DIR_SEP, false)) {
-		for (size_t i = strlen(DIR_SEP); i > 0; --i) {
-			path.pop_back();
-		}
-	}
+	if (path.empty() || PathIsDirectory(path)) return;
 
-	/* Recursively create parent directories. */
-	const size_t sep_pos = path.rfind(DIR_SEP);
-	if (sep_pos != std::string::npos) MakeDirectory(path.substr(0, sep_pos));
-
-#ifdef _WIN32
-	if (CreateDirectory(path.c_str(), NULL)) return;
-	error("Failed creating directory '%s'\n", path.c_str());
-#else
-	if (mkdir(path.c_str(), 0x1FF) == 0) return;
-	error("Failed creating directory '%s' (%s)\n", path.c_str(), strerror(errno));
-#endif
+    bool check = std::filesystem::create_directories(path);
+    if( !check ) {
+	    error("Failed creating directory '%s'\n", path.c_str());
+    }
 }
 
 /**
  * Copy a file.
  * @param src Source file.
  * @param dest Destination file.
  */
-void CopyBinaryFile(const char *src, const char *dest)
+void CopyBinaryFile(const std::string &src, const std::string &dest)
 {
-	FILE *in_file = nullptr;
-	in_file = fopen(src, "rb");
-	if (in_file == nullptr) {
-		error("Could not open file for reading: %s\n", src);
-	}
 
-	FILE *out_file = nullptr;
-	out_file = fopen(dest, "wb");
-	if (out_file == nullptr) {
-		error("Could not open file for writing: %s\n", dest);
+
+	if (src.empty() || !PathIsFile(src)) {
+		error("Path is empty or not a file: %s\n", src.c_str());
 	}
 
-	for (;;) {
-		int byte = getc(in_file);
-		if (byte == EOF) break;
-		putc(byte, out_file);
+	if (dest.empty() || PathIsFile(dest)) {
+		error("Path is empty or not a file: %s\n", dest.c_str());
 	}
 
-	fclose(in_file);
-	fclose(out_file);
+    std::filesystem::copy_file(src,dest);
```

Comment:
Inconsistent whitespace with previous lines (there are other occurrences of this as well).

Reply:
This should be fixed now